Writerside/topics/implementation/public-transit-service.md

```diff
@@ -1,9 +1,166 @@
 # Public Transit Service
 
+In the service layer of our architecture, use cases intersect with the technical details of transit schedules and
+routing. These details are abstracted behind the `PublicTransitService` interface, ensuring that the application only
+interacts with well-defined interfaces without needing to know the underlying complexity. This package has two
+primary objectives:
+
+1. **Stable Service Interface**: To provide a consistent and abstract interface that remains independent of the
+   specific public transit schedule or routing implementation and potentially allows changing the implementation.
+2. **Concrete Service Implementation**: To integrate a concrete data source and routing engine, in our case,
+   the `GtfsRaptorService`, which implements the `PublicTransitService` interface.
+
+The `PublicTransitService` is divided into two responsibilities covered by the `ScheduleInformationService` and
+the `ConnectionRoutingService` interfaces.
+
+```plantuml
+@startuml
+package service {
+    interface PublicTransitService {
+    }
+    interface ScheduleInformationService {
+    }
+    interface ConnectionRoutingService {
+    }
+}
+
+package service.walk {
+    interface WalkCalculator {
+    }
+    class BeeLineWalkCalculator {
+    }
+}
+
+package service.gtfs.raptor {
+    class GtfsRaptorService {
+    }
+}
+
+package gtfs.schedule.model {
+    class GtfsSchedule {
+    }
+}
+
+package raptor {
+    interface RaptorAlgorithm {
+    }
+}
+
+package raptor.router {
+    class RaptorRouter {
+    }
+}
+
+package service.gtfs.raptor.convert {
+    class GtfsToRaptorConverter {
+    }
+    class GtfsRoutePartitioner {
+    }
+}
+
+PublicTransitService <|... GtfsRaptorService: <<implements>>
+PublicTransitService --|> ScheduleInformationService: <<extends>>
+PublicTransitService --|> ConnectionRoutingService: <<extends>>
+
+GtfsRaptorService --> GtfsToRaptorConverter : uses
+GtfsRaptorService *- WalkCalculator : has
+GtfsRaptorService *- GtfsSchedule : has
```

Comment:
Just wondering if we should also add that the GtfsRaptorService creates the Raptor Instance in it's constructor from the Gtfs Schedule

Reply:
Yes you are right, we directly rely on the RaptorRouter and not on the RaptorAlgorithm interface in the GtfsRaptorService. Fixed in: https://github.com/naviqore/documentation/pull/30/commits/fc8274a2f8c22b857bcb42ff72ee988af97f935f